purescript-arrays/Data_Array_ST.go

```diff
@@ -0,0 +1,33 @@
+package purescript_arrays
+
+import . "github.com/purescript-native/go-runtime"
+
+func init() {
+
+	exports := Foreign("Data.Array.ST")
+
+	exports["empty"] = func() Any {
+		return []Any{}
+	}
+
+	exports["pushAll"] = func(as_ Any) Any {
+		return func(xs_ Any) Any {
+			return func() Any {
+				// TODO: buggy implementation, we need to update the reference not create a new slice
+				as := (as_).([]Any)
+				xs := xs_.([]Any)
+				as = append(as, xs...)
```

Comment:
This alone won't fix it (due to the reference issue we've been discussing), but I believe this should be
```go
xs = append(xs, as...)
```